### PR TITLE
Fix #258

### LINKED ISSUE #258: Empty mana bar and its not regenerating.
After getting compendium, I have empty mana bar and its not regenerating. I have even tried to use mana reg. effect such as mana cake, but nothing happened.

### PR DESCRIPTION
**Pull request by IchHabeHunger54**
Fixes #258 which occured due to changes in the magic helper.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,8 +1,14 @@
+# Version 1.18.1-0.1.2
```
Comment (Minecraftschurli) [inline changelog.md]: The .2 version is already live use .3

**Review by Minecraftschurli — CHANGES_REQUESTED** [review]
Why do we even have awardLevel also why does awardXp 0 not work anymore?

**Comment by IchHabeHunger54** [comment]
awardLevel() was intended to be a utility method, originally introduced for the magic xp command. To actually get it work, I needed to tinker around in the magic helper. These changes now cause the awardXp(0) call to add the capability to the player (and therefore display the HUD), but not trigger the first levelup, hence no mana and burnout regen.

**Comment by Minecraftschurli** [comment]
looking at the code awardXp 0 should still work can you explain why it doesn't

**Comment by IchHabeHunger54** [comment]
In MagicHelper#setXp, there is a loop for triggering the levelup events. In the EventHandler class, these levelup events then are used to set the attributes on the player.

However, if you look at line 73 in MagicHelper, the loop is cancelled before any levelup events can be posted to the event bus, due to the xp parameter being 0 (and therefore lower than the required xp for the next level).

MagicHelper#setLevel does not have the problem of needing to worry about the xp for the next level (this is why the method does not call setXp() like it originally used to).

That said, setXp works as it should - except for the level 0->1 scenario, because level 1 is effectively the lowest level and level 0 means the player doesn't know magic at all. Therefore, using setLevel here is the design choice that makes the most sense.

**Comment by Minecraftschurli** [comment]
just change the check in getXpForNextLevel from 0 to 1 then that is a much easier fix

**Comment by IchHabeHunger54** [comment]
Alright. Here you go.
I still left the pickup listener commented out, as I think requiring the player to craft it themselves is the better way.

**Review by Minecraftschurli — CHANGES_REQUESTED** [review]
Update changelog

**Comment by IchHabeHunger54** [comment]
I fixed an additional bug I found.

It seems that JEI does not like brewing recipes that are not subclasses of Forge's BrewingRecipe class (which our AMBrewingRecipe was, that one only implemented IBrewingRecipe).
AMBrewingRecipe has been deleted and usages have been replaced with BrewingRecipe.

To actually get this work, I needed to build a workaround for Forge's NBTIngredient, which is currently bugged (see MinecraftForge/MinecraftForge#8458 for that), which is the newly introduced NBTIngredient class, which simply exposes the constructor of Forge's NBTIngredient (otherwise, only the bugged NBTIngredient.of() call would be usable).

**Review by Minecraftschurli — APPROVED** [review]
:shipit:

**Pull request merged**